### PR TITLE
Make `echo` printing more consistent

### PR DESCRIPTION
**Pull request by IanManske**
# Description
Reverts changes to `echo` from #11934 so that `echo` will always print values if the output stream is the terminal. Printing should also hopefully be more consistent compared to before #11934. For example, take:
```nushell
(echo a; echo b)
```
Before this used to only print `b` , but it now prints both `a` and `b`.

### CODE REVIEW

File: crates/nu-cmd-lang/src/core_commands/echo.rs [hunk 1]
```diff
@@ -38,7 +42,16 @@ little reason to use this over just writing the values as-is."#
...
-        Ok(value.into_pipeline_data())
+        let value = value.into_pipeline_data();
+        match stack.stdout() {
+            IoStream::Pipe | IoStream::Capture => Ok(value),
+            IoStream::Null => Ok(PipelineData::Empty),
+            IoStream::Inherit => {
+                value.print(engine_state, stack, false, false)?;
+                Ok(PipelineData::Empty)
+            }
+            IoStream::File(_) => todo!(),
```
Comment (IanManske) [inline crates/nu-cmd-lang/src/core_commands/echo.rs:53]: Not sure what should happen here. E.g.,
```nushell
echo test o> file.txt | str length
```

**Comment by fdncred** [comment]
wait a second, seems like this is reverting the simpler `echo` where this works like `print` in some circumstances. am i reading that right?

i'd rather have `echo` always output nushell values and `print` always output dumb strings. not sure how others feel though.

**Comment by fdncred** [comment]
No sir. Sorry I wasn't clear. I just wanted to be sure that whatever we were doing before #10338 was included in your latest changes. I didn't see anything wrong, but I wasn't quite sure.

**Comment by IanManske** [comment]
Ohhhh, my two sleepy brain cells have finally comprehended what you mean. But yes, the current behavior on main should match the behavior before #10338.

**Comment by fdncred** [comment]
Thanks @IanManske. I'm also about to pass out from being too tired.

**Pull request closed without merging**